tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,89 @@
+- Enabling cross-utilization of models designed for one modality to work with a different modality. This topic is covered by the tutorials under the [Cross-Modality](./README.md#Cross-Modality) section.    
+
+In the following we will discover the text modality by going through a concrete example involving text auxiliary information. The same principles would apply for the other modalities, when dealing with graph (e.g, social network) or visual (e.g., product images) auxiliary data.
+
+## Example: text auxiliary data
+
+For convenience, assume that we are fitting and evaluating a recommender model leveraging item textual auxiliary data.  As a running example, let's consider the Collaborative Deep Learning ([CDL](../cornac/models/cdl)) model, which is already implemented in Cornac.
+ 
+ 
+### Dataset
+We use the well know MovieLens ML100K dataset. It consists of user-movie interactions in the triplet format `(user_id, movie_id, rating)`, as well as movie plots in the format: `(movie_id, text)`, which represent our textual auxiliary information. This dataset is already accessible from Cornac, and we can load it as follows,
+```Python
+import cornac
+from cornac.data import Reader
+from cornac.datasets import movielens
+
+plots, movie_ids = movielens.load_plot()
+rating_data = movielens.load_100k(reader=Reader(item_set=movie_ids, bin_threshold=3))
+```
+where we have filtered out movies without plots and binarized the integer ratings thanks to the `Reader`.
+
+### The Text Modality class
+
+With our dataset in place, the next step is to instantiate a TextModality, which will allow us to manipulate and represent our auxiliary data in the desired format.  
+ ```Python
+from cornac.data import TextModality
+from cornac.data.text import BaseTokenizer
+
+item_text_modality = TextModality(corpus=plots, ids=movie_ids, 
+                                  tokenizer=BaseTokenizer(sep='\t', stop_words='english'),
+                                  max_vocab=5000, max_doc_freq=0.5)
+```
+In addition the movies plots and ids, we have specified a tokenizer to split text, we limited the maximum vocabulary size to 5000, as well as filtered out words occurring in more than 50% of documents (plots in our case) by setting `max_doc_freq = 0.5`. For more options/details please refer to the [docs](https://cornac.readthedocs.io/en/latest/data.html#module-cornac.data.text). 
+ 
+ 
+**Bag-of-Word text representation.** CDL assumes the bag-of-word representation for text information, i.e., in the form of a document-word matrix. The good news is that we don't have to worry about how to generate such a representation from our raw texts. The TextModality class implements the necessary routines to process and output different representations for text data, e.g., sequence, bag of word. That is, to get our auxiliary data under the desired format, all we need inside the CLD code `cornac/models/cdl/recom_cld` is the following line of code:
```

Comment:
**Bag-of-Word text representation.** CDL assumes the bag-of-word representation for text information, i.e., in the form of a document-word matrix. The good news is that we don't have to worry about how to generate such representation from our raw texts. The `TextModality` class implements necessary routines to process and output different representations for textual data, e.g., text sequences, bag of words, tf-idf. That is, to get our auxiliary data under the desired format, all we need inside the `CDL` model [implementation](https://github.com/PreferredAI/cornac/blob/1177fe6ccfdf5ac582f77bef5aa6f10a31ff9c85/cornac/models/cdl/recom_cdl.py#L162) is the following line of code: